service/src/main/java/org/broadinstitute/listener/relay/Utils.java

```diff
@@ -0,0 +1,25 @@
+package org.broadinstitute.listener.relay;
+
+import static com.google.common.net.HttpHeaders.AUTHORIZATION;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class Utils {
+  public static final String TOKEN_NAME = "LeoToken";
+  public static final String SET_COOKIE_API_PATH = "setCookie";
+
+  public static final Optional<String> getTokenFromAuthorization(Map<String, String> headers) {
+    var authValue = headers.getOrDefault(AUTHORIZATION, null);
+
+    return Optional.ofNullable(authValue)
+        .filter(s -> s.contains("Bearer "))
+        .map(s -> s.replaceFirst("Bearer ", "").trim());
+  }
+
+  public static boolean isSetCookiePath(java.net.URI uri) {
+    var splitted = uri.getPath().split("/");
+    if (splitted.length == 3) return splitted[2].equals(SET_COOKIE_API_PATH);
```

Comment:
Is there a case where you would get http://foo.com, instead of http://foo.com/? If so, the implementation won't work.

Reply:
it'll work...
```
@ java.net.URI.create("https://relay-ns-995014ac-a05d-477e-9a75-1aa10bd5def5.servicebus.windows.net/saturn-4c90d242-dc9d-4d26-b858-4b7c73a97efc/setCookie/").getPath().split("/")
res1: Array[String] = Array("", "saturn-4c90d242-dc9d-4d26-b858-4b7c73a97efc", "setCookie")

vs

@ java.net.URI.create("https://relay-ns-995014ac-a05d-477e-9a75-1aa10bd5def5.servicebus.windows.net/saturn-4c90d242-dc9d-4d26-b858-4b7c73a97efc/setCookie").getPath().split("/")
res3: Array[String] = Array("", "saturn-4c90d242-dc9d-4d26-b858-4b7c73a97efc", "setCookie")
```

But I'll add a unit test as well

Reply:
Okay, I misunderstood the method, what is happening here is a check if we are dealing with a set cookie request.  Adding the unit tests would be good.